publication-doi-commons/src/main/java/no/unit/nva/publication/doi/dynamodb/dao/DynamodbStreamRecordJsonPointers.java

```diff
@@ -2,48 +2,111 @@
-        "/dynamodb/newImage/entityDescription/m/date/m/month/s");
-    public static final JsonPointer ENTITY_DESCRIPTION_PUBLICATION_DATE_DAY = JsonPointer.compile(
-        "/dynamodb/newImage/entityDescription/m/date/m/day/s");
-
-    public static final JsonPointer MAIN_TITLE_POINTER = JsonPointer.compile(
-        "/dynamodb/newImage/entityDescription/m/mainTitle/s");
-    public static final JsonPointer TYPE_POINTER = JsonPointer.compile("/dynamodb/newImage/type/s");
-
-    public static final JsonPointer DOI_POINTER = JsonPointer.compile(
-        "/dynamodb/newImage/entityDescription/m/reference/m/doi/s");
-
-    public static final String CONTRIBUTOR_POINTER = "/dynamodb/newImage/entityDescription/m/contributors";
-    public static final JsonPointer CONTRIBUTORS_LIST_POINTER = JsonPointer.compile(
-        "/dynamodb/newImage/entityDescription/m/contributors/l");
-    public static final JsonPointer CONTRIBUTOR_ARP_ID_JSON_POINTER = JsonPointer.compile("/m/identity/m/arpId/s");
-    public static final JsonPointer CONTRIBUTOR_ORC_ID_JSON_POINTER = JsonPointer.compile("/m/identity/m/orcId/s");
-    public static final JsonPointer CONTRIBUTOR_NAME_JSON_POINTER = JsonPointer.compile("/m/identity/m/name/s");
+    private static final String PUBLISHER_ID = "/publisherId/s";
+
+    private static final String ENTITY_DESCRIPTION_REFERENCE_TYPE =
+        "/entityDescription/m/reference/m/publicationInstance/m/type/s";
+    private static final String ENTITY_DESCRIPTION_MAP = "/entityDescription/m";
+    private static final String ENTITY_DESCRIPTION_PUBLICATION_DATE_YEAR = "/entityDescription/m/date/m/year/s";
+    private static final String ENTITY_DESCRIPTION_PUBLICATION_DATE_MONTH = "/entityDescription/m/date/m/month/s";
+    private static final String ENTITY_DESCRIPTION_PUBLICATION_DATE_DAY = "/entityDescription/m/date/m/day/s";
+
+    private static final String ENTITY_DESCRIPTION_MAIN_TITLE = "/entityDescription/m/mainTitle/s";
+    private static final String TYPE_POINTER = "/type/s";
+
+
+    private static final String ENTITY_DESCRIPTION_REFERENCE_DOI = "/entityDescription/m/reference/m/doi/s";
+
+    private static final String ENTITY_DESCRIPTION_CONTRIBUTORS = "/entityDescription/m/contributors";
+    private static final String ENTITY_DESCRIPTION_CONTRIBUTORS_LIST = "/entityDescription/m/contributors/l";
+    private static final String IDENTITY_ARP_ID = "/m/identity/m/arpId/s";
+    private static final String IDENTITY_ORC_ID = "/m/identity/m/orcId/s";
+    private static final String IDENTITY_NAME = "/m/identity/m/name/s";
+
+    public static final JsonPointer TYPE_JSON_POINTER = JsonPointer.compile("/type/s");
 
     public static final String DYNAMODB_TYPE_STRING = "s";
     public static final String DYNAMODB_TYPE_LIST = "l";
 
-    private DynamodbStreamRecordJsonPointers() {
+    private final String base;
+
+    public DynamodbStreamRecordJsonPointers(String base) {
+        this.base = base;
```

Comment:
Construct all the JsonPointer in the constructor instead of every time we call a getter? I'm thinking of this is kinda the same as when you compile a regex, you only want to do it once as it is a bit compute expensive?